### PR TITLE
Opcua 1708 reorganize how pulls log it

### PR DESCRIPTION
**Pull request by pnikiel**
* before it, the mode of building against "external" LogIt didn't work,
* also added more validators on the user-supplied values plus puts a nice message if the LogIt external library was not found
* also spawned whole LogIt resolution code from master CMakeLists into a separate file for easier maintenance

### CODE REVIEW

**Review by parasxos — APPROVED** [review]

**Comment by ben-farnham** [comment]
Hi Piotr,

Sorry I'm slow - Tuesday: traditionally the day of meetings. I'll try and take a look later this afternoon (last meeting starts 16h00, hopefully not longer than 1hr). If I fail then definitely tomorrow morning.

Cheers,
Ben

File: LogIt.cmake [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  # Parse LOGIT_BUILD_OPTION first as it determines almost any other behaviour that follows
+  # Possible options:
+  # - Source          --> fetches LogIt into ${PROJECT_BINARY_DIR}/LogIt
+  # - External shared --> uses existing LogIt, its headers should be in ${LOGIT_INCLUDE_DIR}
+  # - External static --> uses existing LogIt, its headers should be in ${LOGIT_INCLUDE_DIR}
+  # If nothing coming from outside, make "LOGIT_AS_INT_SRC" the default
+  set(LOGIT_BUILD_OPTION "LOGIT_AS_INT_SRC" CACHE STRING "LogIt is a mandatory dependency of open62541-compat. Inclusion options LOGIT_AS_INT_SRC, LOGIT_AS_EXT_SHARED, LOGIT_AS_EXT_STATIC")
+  set(LOGIT_BUILD_OPTIONS_SUPPORTED "LOGIT_AS_INT_SRC" "LOGIT_AS_EXT_SHARED" "LOGIT_AS_EXT_STATIC")
+  
+  if (NOT "${LOGIT_BUILD_OPTION}" IN_LIST LOGIT_BUILD_OPTIONS_SUPPORTED)
```
Comment (ben-farnham) [inline LogIt.cmake:50]: IN_LIST is nice

File: LogIt.cmake [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+# GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors:
+# Ben Farnham <firstNm.secondNm@cern.ch>
+# Piotr Nikiel <piotr@nikiel.info>
+# 
+# Note: the following code was mostly contributed by Ben, in the master CMakeLists of open62541-compat
+# Then moved by Piotr to this file for easier maintenance, further fixes by Piotr
```
Comment (ben-farnham) [inline LogIt.cmake:22]: moving this out is a good idea

File: LogIt.cmake (hunk 1 above)
Comment (ben-farnham) [reply LogIt.cmake:50]: suggest adding actual build option provided to error message to help user identify what went wrong
```
message(FATAL_ERROR "Given LOGIT_BUILD_OPTION [${LOGIT_BUILD_OPTION}] is not supported. It must be one of: ${LOGIT_BUILD_OPTIONS_SUPPORTED}")
```

Can help to quickly identify what went wrong where
```
// input
cmake ../open62541-compat -DSTANDALONE_BUILD=ON -DOPEN62541-COMPAT_BUILD_CONFIG_FILE=boost_custom_cc7.cmake -DCMAKE_BUILD_TYPE=DEBUG -DLOGIT_BUILD_OPTION=WTF

// outut
CMake Error at LogIt.cmake:51 (message):
  Given LOGIT_BUILD_OPTION [WTF] is not supported. etc etc
```

**Review by ben-farnham — COMMENTED** [review]
Yep generally looks good to me (1 suggestion made in review: fairly trivial).

Made (admittedly a little perfunctory) windows test: built stand-alone and executed unit tests. All good.

File: LogIt.cmake (hunk 1 above)
Comment (pnikiel) [reply LogIt.cmake:50]: thanks, done.

**Pull request merged**